ql/src/java/org/apache/hadoop/hive/ql/io/parquet/convert/HiveStructConverter.java

```diff
@@ -96,6 +96,34 @@ private void init(final GroupType selectedGroupType,
     }
   }
 
+  private boolean findSubType(
+    final GroupType groupType,
+    final Type subtype) {
+    if (subtype.isPrimitive()) {
+      return groupType.getFields().contains(subtype);
+    } else if(subtype.isRepetition(Type.Repetition.REPEATED)){
+      //TODO support isList case
+      return groupType.getFields().contains(subtype);
+    }else {
+      for (Type g : groupType.getFields()) {
+        if (!g.isPrimitive() && g.getName().equals(subtype.getName())) {
+          // check all elements are contained in g
+          boolean containsAll = false;
+          for (Type subSubType : subtype.asGroupType().getFields()) {
```

Comment:
This method is recursively checking whether a type is a subtype for the group type.